Fixes #1655. Removed explicit field references.

Ichabond opened the pull request:
### Fixes:
Fixes #1655

All .name references are now removed where applicable (eg. not in the detailed description, as that's an explicit UI reference to .name, so might not work well with `__str__`). Updated the `Secrets` `__str__` method, as this was lacking a reference to `name`, which felt oddly lacking.

Ichabond committed:
Fixes #1655

Further field name references were found in `consoleport.html`. These have now been removed, so we rely on proper a proper `__str__` implementation of both `ConsolePort` and `ConsoleServerPort`.

Ichabond committed:
Fixes #1655: Removed explicit field references

Cleaned up all (notable) .name references, and removed them so __str__ can do the lifting. Did not remove the references where it was explicitly referenced to .name (eg. in details). Extended the Secret model to also include the name in __str__, since that was weirdly absent.

jeremystretch commented on netbox/templates/secrets/inc/secret_tr.html:
```diff
@@ -1,7 +1,7 @@
 {% load secret_helpers %}
 <tr>
     <td><a href="{% url 'secrets:secret' pk=secret.pk %}">{{ secret.role }}</a></td>
-    <td>{{ secret.name }}</td>
```
We specifically want to only show the name here (this is a column in a table).

jeremystretch commented on netbox/secrets/models.py:
```diff
@@ -286,8 +286,8 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
-        if self.role and self.device:
-            return '{} for {}'.format(self.role, self.device)
+        if self.role and self.device and self.name:
```
We still want to return the role and device if no name is set.

jeremystretch commented on netbox/templates/dcim/inc/rack_elevation.html:
```diff
@@ -27,13 +27,13 @@
                         <a href="{% url 'dcim:device' pk=u.device.pk %}" data-toggle="popover" data-trigger="hover" data-container="body" data-html="true"
                            data-content="{{ u.device.device_role }}<br />{{ u.device.device_type.full_name }} ({{ u.device.device_type.u_height }}U){% if u.device.asset_tag %}<br />{{ u.device.asset_tag }}{% endif %}">
-                            {{ u.device.name|default:u.device.device_role }}
+                            {{ u.device|default:u.device.device_role }}
```
We want to look specifically at `name` in this case, because `__str__` calls `display_name()`, which will return the device type if no name is set.

jeremystretch commented on netbox/templates/dcim/inc/rack_elevation.html:
```diff
@@ -27,13 +27,13 @@
                             {% endif %}
                         </a>
                     {% else %}
-                        <span>{{ u.device.name|default:u.device.device_role }}</span>
```
Same as above.

jeremystretch commented on netbox/templates/dcim/rack_elevation_list.html:
```diff
@@ -14,7 +14,7 @@
                 {% for rack in page %}
                     <div style="display: inline-block; width: 266px">
                         <div class="rack_header">
-                            <strong><a href="{% url 'dcim:rack' pk=rack.pk %}">{{ rack.name|truncatechars:"25" }}</a></strong>
```
As with devices, we intentionally reference `name` here to avoid calling `display_name()`.

jeremystretch commented on netbox/templates/dcim/rack_elevation_list.html:
```diff
@@ -24,7 +24,7 @@
                         {% endif %}
                         <div class="clearfix"></div>
                         <div class="rack_header">
-                            <strong><a href="{% url 'dcim:rack' pk=rack.pk %}">{{ rack.name|truncatechars:"25" }}</a></strong>
```
Same as above.

jeremystretch commented on netbox/templates/ipam/inc/service.html:
```diff
@@ -1,5 +1,5 @@
 <tr>
-    <td>{{ service.name }}</td>
```
We want only the service name here.

jeremystretch reviewed (changes requested):
Most of these are fine, but there are several instances where `name` is referenced intentionally.

Ichabond committed:
Adapted PR to comply with comments

Re-introduced certain references to make sure explicit references are still used where needed.

Ichabond commented:
I've gone over them, and I agree. I've pushed a new commit reverting those changes explicitly mentioned. Thanks for the feedback on this.

jeremystretch commented:
LGTM
